pkg/logs/parser/parser.go

```diff
@@ -50,12 +50,17 @@ type DecodingParser struct {
 	Parser
```

Comment:
Unless I'm missing something I don't think we need to embed the `Parser` interface in the `DecodingParser` struct (same for `noopParser`), and if we didn't the compiler would catch such issues.

Reply:
I agree, and it's not that big of a change, as per subsequent commit, I think we can include it in 7.23 WDYT ?